addons/l10n_in/models/company.py

```diff
@@ -174,6 +175,26 @@ def _update_l10n_in_is_gst_registered(self):
             if company.country_code == "IN" and company.vat:
                 company.l10n_in_is_gst_registered = company.partner_id.check_vat_in(company.vat)
 
+    def _load_l10n_in_branch_journals(self):
+        """Load journals for Indian branches having different GSTIN than parent company."""
+
+        # Process only Indian branches with GST and a parent company.
+        for branch in self.filtered(lambda b: (
+            b.parent_id
+            and b.country_code == 'IN'
+            and b.partner_id.check_vat_in(b.vat)
+            and b.parent_id.country_code == 'IN'
```

Comment:
```suggestion
            and {b.country_code, b.parent_id.country_code} == {'IN'}
            and self.env['res.partner'].check_vat_in(b.vat)
```
We can do a single country check by using a set and preferred self.env because the method doesn't depend on the object